src/core/MIGRATION.md

```diff
@@ -1143,3 +1146,26 @@ export const myPluginMocks = {
   createStart: ...
 }
 ```
+Plugin mocks should consist of mocks for *public APIs only*: setup/start/stop contracts. Mocks aren't necessary for pure functions as other plugins can call the original implementation in tests.
+
+#### Using mocks in your tests
+During the migration process, it is likely you are preparing your plugin by shimming in new platform-ready dependencies via the legacy `ui/new_platform` module:
+```typescript
+import { npSetup, npStart } from 'ui/new_platform';
+```
+
+If you are using this approach, the easiest way to mock core and new platform-ready plugins in your legacy tests is to mock the `ui/new_platform` module:
+```typescript
+jest.mock('ui/new_platform');
```

Comment:
We should add a note that one should use `jest.mock('ui/new_platform');` only for old Jest tests.

If you are writing new tests, you should structure your code and tests such that you don't need this mock.

Instead you should import the `core` mock directly and instantiate it.